packer/main.pkr.hcl

```diff
@@ -8,61 +8,106 @@ packer {
 }
 
 source "googlecompute" "ubuntu-2204" {
-  project_id          = "containerssh"
+  project_id          = var.project_id
   source_image_family = "ubuntu-pro-2204-lts"
   ssh_username        = "root"
   zone                = "europe-west3-c"
   account_file        = "./gcp.key.json"
+  machine_type        = "e2-small"
 }
 
 build {
   name = "sacrificial-vm-image"
-
   source "googlecompute.ubuntu-2204" {
     image_name = "sacrificial-vm-image"
   }
 
+  provisioner "shell" {
+    inline = [
+      "mkdir -p /home/deployer/tmp",
+      "mkdir /etc/docker/",
+      "mkdir /var/docker/"
+    ]
+  }
+
   provisioner "file" {
-    source      = "./scripts/util_fn"
-    destination = "/tmp/util_fn"
+    source      = "./files/ca_server.tar"
+    destination = "/home/deployer/tmp/ca_server.tar"
   }
 
+  provisioner "file" {
+    source      = "./scripts/apt_get_wait_lock.sh"
+    destination = "/home/deployer/tmp/apt_get_wait_lock.sh"
+  }
   provisioner "shell" {
-    script = "./scripts/update.sh"
+    script            = "./scripts/update_apt_packages.sh"
+    expect_disconnect = true
   }
-
   provisioner "file" {
-    source      = "./scripts/util_fn"
-    destination = "/tmp/util_fn"
+    source      = "./scripts/apt_get_wait_lock.sh"
+    destination = "/home/deployer/tmp/apt_get_wait_lock.sh"
   }
+
   provisioner "shell" {
     script = "./scripts/install_docker.sh"
   }
+
+  provisioner "shell" {
+    inline = [
+      "docker pull containerssh/containerssh-guest-image:latest"
+    ]
+  }
 }
 
 build {
-  name = "logger-vm-image"
-
+  name = "ubuntu-with-docker-image"
   source "googlecompute.ubuntu-2204" {
-    image_name = "logger-vm-image"
+    image_name = "ubuntu-with-docker-image"
+  }
+
+  provisioner "shell" {
+    inline = [
+      "mkdir -p /home/deployer/tmp",
+    ]
+  }
+
+  # test
+  provisioner "file" {
+    source      = "./files/ca_client.tar"
+    destination = "/home/deployer/tmp/ca_client.tar"
```

Comment:
We probably shouldn't commit these files for security.
Will remove relevant parts and  use for manual configuration now.

We could automate it later using Ansible

Reply:
removed